docs/guides/themeable-to-emotion-migration-guide.md

```diff
@@ -0,0 +1,459 @@
+
+Create a `styles.js` file next to the `theme.js`. (Don't forget the copyright block!)
+
+Write and export a function named `generateStyle`. It receives the `componentTheme` generated by `theme.js`, the component props and a `state` object as arguments. These additional `state` variables can be passed from the component in the `makeStyles` method (see later).
+
+Use the content of the `styles.css` and convert it to css-in-js. Use the passed component variables, props and state where needed. (More info in the [emotion](#emotion) docs.) Use [Emotion's Object Styles documentation](https://emotion.sh/docs/object-styles) or other InstUI components as a reference.
+
+```js
+/**
+ * Generates the style object from the theme and provided additional information
+ * @param  {Object} componentTheme The theme variable object.
+ * @param  {Object} props the props of the component, the style is applied to
+ * @param  {Object} state the state of the component, the style is applied to
+ * @return {Object} The final style object, which will be used in the component
+ */
+const generateStyle = (componentTheme, props, state) => {
+  const { color } = props
+
+  // optional mappings can be provided based on props
+  const colorStyles = {
+    primary: {
+      color: componentTheme.primaryColor,
+      fontSize: '20px'
+    },
+    info: {
+      color: componentTheme.infoColor
+    }
+  }
+
+  // return with the css you'd like to apply to the component
+  return {
+    componentName: {
+      label: 'componentName',
+      display: 'block',
+      background: componentTheme.background,
+      boxSizing: 'border-box',
+      ...colorStyles[color]
+    },
+    child: {
+      label: 'componentName__child',
+      color: 'green',
+      background: componentTheme.infoColor
+    }
+  }
+}
+export default generateStyle
+```
+
+#### 3. Make changes in the component
+
+In the `index.js` of the component, replace the imports, the decorator, and refactor the style handling.
+
+##### 1. jsx
+
+Add the `/** @jsx jsx */` annotation on top.
+
+```js
+/** @jsx jsx */
+import { Children, Component } from 'react'
+```
+
+##### 2. imports
+
+- Replace `@instructure/ui-themeable` import with `{ withStyle, jsx }` from `@instructure/emotion`.
+- Replace the `theme` import with component theme generator (`generateComponentTheme`) from `theme.js`.
+- Replace the `styles.css` import with the style generator (`generateStyle`) import from `styles.js`.
+- Don't forget to update the dependencies in `package.json`.
+
+Note: these utils were moved from `ui-themeable` to the `emotion` package: ThemeablePropValues, ThemeablePropTypes, makeThemeVars, getShorthandPropValue, mirrorShorthandCorners, mirrorShorthandEdges. Update the import where needed.
```

Comment:
We should make a codemod for this step, I've created INSTUI-2892

Reply:
Good idea! Thx.